kubetest2/internal/deployers/eksapi/infra.go

```diff
@@ -80,95 +82,210 @@ func (i *Infrastructure) subnets() []string {
 	return append(i.subnetsPublic, i.subnetsPrivate...)
 }
 
-func (m *InfrastructureManager) createInfrastructureStack(opts *deployerOptions) (*Infrastructure, error) {
+func (m *InfrastructureManager) getOrCreateInfrastructureStack(opts *deployerOptions) (*Infrastructure, error) {
 	publicKeyMaterial, err := loadSSHPublicKey()
 	if err != nil {
 		return nil, err
 	}
-	// get two AZs for the subnets
-	azs, err := m.clients.EC2().DescribeAvailabilityZones(context.TODO(), &ec2.DescribeAvailabilityZonesInput{})
-	if err != nil {
-		return nil, err
-	}
-	var subnetAzs []string
-	if opts.CapacityReservation {
-		subnetAzs, err = m.getAZsWithCapacity(opts)
+	var infra *Infrastructure
+	if opts.StaticClusterName != "" {
+		klog.Infof("getting infrastructure resource for static cluster: %v", opts.StaticClusterName)
+		describeOutput, describeErr := m.clients.EKS().DescribeCluster(context.TODO(), &eks.DescribeClusterInput{
+			Name: &opts.StaticClusterName,
+		})
+		if describeErr != nil {
+			return nil, fmt.Errorf("failed to describe static cluster: %v", describeErr)
+		}
+		sshkeyPair, err := m.getOrCreateSSHKeyPair(publicKeyMaterial)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create SSH KeyPair: %v", err)
+		}
+		vpc := *describeOutput.Cluster.ResourcesVpcConfig.VpcId
+		sshSecurityGroup, err := m.getOrCreateSSHSecurityGroup(vpc)
+		if err != nil {
```

Comment:
could we assume the sg must exist? theres a default in EKS cluster creation right?

Reply:
ack. reusing `*describeOutput.Cluster.ResourcesVpcConfig.ClusterSecurityGroupId`